### PR TITLE
CompatHelper: bump compat for "StaticArrays" to "1.0"

### PR DESCRIPTION
**Pull request by sethaxen**
This pull request changes the compat entry for the `StaticArrays` package from `0.12` to `0.12, 1.0`.

This keeps the compat entries for earlier versions.

Note: I have not tested your package with this new compat entry. It is your responsibility to make sure that your package tests pass before you merge this pull request.

### CODE REVIEW

**Comment by mateuszbaran** [comment]
Now it's blocked by https://github.com/JuliaImages/ImageFiltering.jl/pull/196

**Comment by mateuszbaran** [comment]
Now it looks like we can't even use Julia 1.3 since SparseDiffTools requires 1.4 :confused: .

**Comment by mateuszbaran** [comment]
@kellertuer what do you think about merging it with the changes I've made?

File: docs/src/misc/internals.md [hunk 1]
```diff
@@ -5,7 +5,6 @@ This page documents the internal types and methods of `Manifolds.jl`'s that migh
 ## Types
```
Comment (kellertuer) [inline docs/src/misc/internals.md:5]: We could remove all lines 5-10?

**Review by kellertuer — COMMENTED** [review]
Looks nice, I just have one small remark.

File: docs/src/misc/internals.md (hunk 1 above)
Comment (mateuszbaran) [reply docs/src/misc/internals.md:5]: Yes, I'll remove them.

**Review by kellertuer — APPROVED** [review]

**Pull request merged**